### ISSUE TITLE
No Data After Triggering API Request, Everything is Zero or Unknown (New Install)

### ISSUE THREAD

**Issue by poolitman** [oziee/ha-solcast-solar#217]
### The problem

I have followed the instructions for the installation of this integration on Github.
I also added the suggested automation for the API Solcast update.
The automation runs (both manually & automatically) but still everything is zero/unknown.
The data on the solcast website appears correctly.
I've tried;  waiting a few days, copying the api key again, uninstalling the integration (both from HA & HACS),
I can see lots of things it says are issues but I do not understand enough to fix them.
Hopefully I have I missed something obvious and its easy to spot & fix (i.e. its just me being thick)?
Sorry if this is obvious.

### What version of Solcast Integration has the issue?

v4.09

### What version of Home Assistant are you using?

2023.10.3

### Diagnostics information

[home-assistant_solcast_solar_2023-10-15T18-52-32.390Z.log](https://github.com/oziee/ha-solcast-solar/files/12910426/home-assistant_solcast_solar_2023-10-15T18-52-32.390Z.log)

### Logs debug/info/error output

```txt
2023-10-15 19:31:29.576 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.577 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.577 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration octopus_energy which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.577 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration myenergi which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.578 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration solax_modbus which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.578 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration meross_lan which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:29.579 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration solis which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-10-15 19:31:37.894 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry F5364-TalkTalkHub for upnp
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 399, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/upnp/__init__.py", line 85, in async_setup_entry
    device = await async_create_device(hass, location)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/upnp/device.py", line 56, in async_create_device
    upnp_device = await factory.async_create_device(ssdp_location)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/async_upnp_client/client_factory.py", line 72, in async_create_device
    root_el = await self._async_get(description_url)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/async_upnp_client/client_factory.py", line 397, in _async_get
    raise UpnpResponseError(status=status_code, headers=response_headers)
async_upnp_client.exceptions.UpnpResponseError: Did not receive HTTP 200 but 403
2023-10-15 19:31:38.189 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_electricity_import_today (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.189 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_electricity_import_this_week (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.190 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_electricity_import_this_month (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.190 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_electricity_import_unit_rate (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.190 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_electricity_import_standing_charge (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.191 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_gas_import_today (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.191 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_gas_import_this_week (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.192 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_gas_import_this_month (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.192 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_gas_import_unit_rate (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.192 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.smart_meter_gas_import_standing_charge (<class 'homeassistant.components.mqtt.sensor.MqttSensor'>) is using state class 'measurement' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mqtt%22
2023-10-15 19:31:38.220 WARNING (MainThread) [custom_components.meross_lan] MerossCloudProfile(2383731): Meross cloud api reported new devices but MQTT publishing is disabled: skipping automatic discovery
2023-10-15 19:31:40.655 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Canon TR4500 series' for ipp integration not ready yet: Invalid response from API: Error occurred while communicating with IPP server.; Retrying in background
2023-10-15 19:31:50.542 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_gas_g4f00274402000_7454960102_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
2023-10-15 19:31:50.612 WARNING (Recorder) [homeassistant.components.recorder.db_schema] State attributes for sensor.octopus_energy_gas_g4f00274402000_7454960102_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
2023-10-15 19:33:00.428 WARNING (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_data Solcast.com http status Error 404 - Gathering rooftop sites data.
2023-10-15 19:33:00.430 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_data Exception error: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 134, in sites_data
    raise Exception(f"SOLCAST - HTTP sites_data error: Solcast Error gathering rooftop sites data.")
Exception: SOLCAST - HTTP sites_data error: Solcast Error gathering rooftop sites data.

2023-10-15 19:33:00.993 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_usage error: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 172, in sites_usage
    raise Exception(f"SOLCAST - sites_usage: gathering site data failed. request returned Status code: {status} - Responce: {resp_json}.")
Exception: SOLCAST - sites_usage: gathering site data failed. request returned Status code: 404 - Responce: {'response_status': {'error_code': 'NotFound', 'message': 'ApiKey does not exist', 'errors': []}}.

2023-10-15 19:33:53.759 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_usage error: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 172, in sites_usage
    raise Exception(f"SOLCAST - sites_usage: gathering site data failed. request returned Status code: {status} - Responce: {resp_json}.")
Exception: SOLCAST - sites_usage: gathering site data failed. request returned Status code: 404 - Responce: {'response_status': {'error_code': 'NotFound', 'message': 'ApiKey does not exist', 'errors': []}}.

2023-10-15 19:35:10.353 WARNING (Recorder) [homeassistant.components.sensor.recorder] The unit of sensor.octopus_energy_gas_g4f00274402000_7454960102_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
2023-10-15 19:52:17.799 INFO (MainThread) [custom_components.solcast_solar] SOLCAST - Service call: update_forecasts
2023-10-15 19:52:17.799 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - Polling API.
2023-10-15 19:52:17.799 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - getting usage data from solcast
2023-10-15 19:52:18.446 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST - sites_usage error: Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/solcastapi.py", line 172, in sites_usage
    raise Exception(f"SOLCAST - sites_usage: gathering site data failed. request returned Status code: {status} - Responce: {resp_json}.")
Exception: SOLCAST - sites_usage: gathering site data failed. request returned Status code: 404 - Responce: {'response_status': {'error_code': 'NotFound', 'message': 'ApiKey does not exist', 'errors': []}}.
```

**Comment by poolitman** [comment]
Sorry I forgot to add I double checked the settings on the solcast site, as I read in previous issues that someone had entered the wrong figures for AC & DC capacity (so the figures produced were to low & therefore showing as zero).

**Comment by oziee** [comment]
In the log it says

`Exception: SOLCAST - sites_usage: gathering site data failed. request returned Status code: 404 - Responce: {'response_status': {'error_code': 'NotFound', 'message': 'ApiKey does not exist', 'errors': []}}.`

API KEY does not exist.. check your api key, make sure it's correct and no spaces or something

**Comment by poolitman** [comment]
Hi
Thanks for the speedy response.
I've entered the api key (in configuration) a number of times, using both the copy url button & using copy & paste (making sure there wasn't a space at the end of the key).
As a relatively new user of solcast I only get 10 requests a day & I didn't turn on debug logging until just before I reported the issue, so I wonder if you get the same error message when you've used up your daily allocation?
I'll try again tomorrow morning (using debug logging) & look for the same error. If I still get it I re-enter the key & if that doesn't work I've read somewhere solcast lets you generate a new api key, I'll try that.
I'll report back whether or not I'm successful.
Thanks for your help.
